core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1284,6 +1274,7 @@ class KafkaApis(val requestChannel: RequestChannel,
                                        topicNames : util.Map[Uuid, String],
                                        sharePartitionManager : SharePartitionManager,
                                        shareFetchContext : ShareFetchContext,
+                                       unauthorizedTopicPartitions: mutable.Set[TopicIdPartition]
                                       ): ShareFetchResponse = {
 
     val shareFetchRequest = request.body[ShareFetchRequest]
```

Comment:
Shouldn't we have common method to handle acknowledgement from ShareFetch and ShareAcknowledge as most of the functionality would remain same?